projectroles/forms.py

```diff
@@ -541,9 +554,88 @@ def __init__(self, project=None, current_user=None, *args, **kwargs):
             self.fields['description'].widget = forms.HiddenInput()
             self.fields['readme'].widget = forms.HiddenInput()
 
+    @classmethod
+    def _validate_app_settings(
+        self,
+        cleaned_data,
+        app_plugins,
+        app_settings,
+        p_kwargs,
+        instance,
+        instance_owner_as,
+    ):
+        """Validate and clean app_settings form fields"""
+        errors = []
+
+        if cleaned_data.get('type') == PROJECT_TYPE_CATEGORY:
+            return (cleaned_data, errors)
+
+        for plugin in app_plugins + [None]:
+            if plugin:
+                name = plugin.name
+                p_settings = app_settings.get_definitions(
+                    APP_SETTING_SCOPE_PROJECT, app_name=name, **p_kwargs
+                )
+            else:
+                name = 'projectroles'
+                p_settings = app_settings.get_definitions(
+                    APP_SETTING_SCOPE_PROJECT, app_name=name, **p_kwargs
+                )
+
+            for s_key, s_val in p_settings.items():
+                s_field = 'settings.{}.{}'.format(name, s_key)
+
+                try:
+                    app_settings_errors = plugin.validate_app_settings(
+                        p_settings,
```

Comment:
We are still providing app setting definitions here, when we should instead be providing the actual values input by the user into the form so they can be validated. The definitions are known by the plugin so they do not need to be provided to this method. Please go back and read what I wrote in the previous review.